snuba/web/views.py

```diff
@@ -285,58 +285,69 @@ def dataset_query_view(*, dataset_name: str, timer: Timer):
         assert False, "unexpected fallthrough"
 
 
-def dataset_query(dataset, body, timer):
+def dataset_query(dataset: Dataset, body, timer: Timer) -> Response:
     assert http_request.method == "POST"
     ensure_table_exists(dataset)
-
-    schema = RequestSchema.build_with_extensions(
-        dataset.get_extensions(), HTTPRequestSettings
-    )
-    query_result = run_query(
-        dataset,
-        validate_request_content(body, schema, timer, dataset, http_request.referrer),
-        timer,
+    return format_result(
+        run_query(
+            dataset,
+            validate_request_content(
+                body,
+                RequestSchema.build_with_extensions(
+                    dataset.get_extensions(), HTTPRequestSettings
+                ),
+                timer,
+                dataset,
+                http_request.referrer,
+            ),
+            timer,
+        )
     )
 
+
+def run_query(dataset: Dataset, request: Request, timer: Timer) -> QueryResult:
+    try:
+        return QueryResult(
+            {
+                **parse_and_run_query(dataset, request, timer),
+                "timing": timer.for_json(),
+            },
+            200,
+        )
+    except RawQueryException as e:
+        return QueryResult(
+            {
+                "error": {"type": e.err_type, "message": e.message, **e.meta},
+                "sql": e.sql,
+                "stats": e.stats,
+                "timing": timer.for_json(),
+            },
+            429 if e.err_type == "rate-limited" else 500,
+        )
+
+
+def format_result(result: QueryResult) -> Response:
     def json_default(obj):
         if isinstance(obj, datetime):
             return obj.isoformat()
         elif isinstance(obj, UUID):
             return str(obj)
         return obj
 
-    return (
-        json.dumps(query_result.result, for_json=True, default=json_default),
```

Comment:
Do you know what we were using this for ?

Reply:
Which "this" — the `for_json`?

Assuming that was what you were intending, It is a `simplejson` proprietary option:

> If for_json is true (not the default), objects with a for_json() method will use the return value of that method for encoding as JSON instead of the object.

The only object that implements this is the `Timer`: https://github.com/getsentry/snuba/blob/e66bda05ec34eaa41a335c67517058c72b1bcbfd/snuba/utils/metrics/timer.py#L58